src/utils.ts

```diff
@@ -206,3 +206,46 @@ export const addTokenEventListener = async (
     contractAddress,
     ClientType.TokenClient,
   );
+
+/**
+ * Extension specific event listener
+ * It creates a new interface with ABI data describing
+ * possible event signature
+ */
+export const addExtensionEventListener = async (
+  eventSignature: ContractEventsSignatures,
+  extensionAddress: string,
+): Promise<void> => {
+  const { provider } = networkClient;
+  const iface = new utils.Interface([`event ${eventSignature}`]);
```

Comment:
I feel this is hacky, but I didn't have a better idea. Open to suggestions on improvements

Reply:
I'm ok with this, but why `Interface` and not just instantiating a new `Contract` instance ?